azure-kusto-ingest/azure/kusto/ingest/_resource_manager.py

```diff
@@ -88,7 +93,9 @@ def _get_resource_by_name(self, table: KustoResultTable, resource_name: str):
         return [_ResourceUri.parse(row["StorageRoot"]) for row in table if row["ResourceTypeName"] == resource_name]
 
     def _get_ingest_client_resources_from_service(self):
-        table = self._kusto_client.execute("NetDefaultDB", ".get ingestion resources").primary_results[0]
+        table = func_retry_wrapper(
+            partial(self._kusto_client.execute, "NetDefaultDB", ".get ingestion resources"), self._max_retry, self._seconds_between_retry, KustoThrottlingError
```

Comment:
Personally, I think a small lambda is more readable than a partial, and either way I'd separate the first arguments from the rest for readability.